app/models/comment.rb

```diff
@@ -7,11 +7,15 @@ class Comment < ApplicationRecord
   attribute :text, :text
 
   # Callbacks
-  after_create :update_post_comments_counter
-  after_destroy :update_post_comments_counter
+  after_create :increase_post_comments_counter
+  after_destroy :decrement_post_comments_counter
 
   # Methods
-  def update_post_comments_counter
-    post.update(comments_counter: post.comments.count)
+  def increase_post_comments_counter
+    post.increment!(:comments_counter)
+  end
+
+  def decrement_post_comments_counter
+    post.decrement!(:comments_counter)
   end
 end
```

Comment:
- [ ] The following change should be done in the `index` action of the `Posts` Controller and anywhere that needs the **_N+1 query_** corrected:

- Kindly correct the **_N+1 query_** in the controller. Also, the screenshot that you took should be that of the terminal logs before correcting the **_N+1 query_** and after correcting the **_N+1 query_**. Kindly fix this and attach the correct screenshot as it is part of this milestone's requirements. 👍 🖱️

Reply:
I don't really understand how or why should I add callbacks to the controller. However, I updated the controllers and views to fairly improve the N+1 problems